docs/site/Inside-Loopback-Application.md

```diff
@@ -6,81 +6,146 @@ sidebar: lb4_sidebar
+  - The mobile/web app submits the order and resets the shopping cart.
+  - User takes a look at all the previous orders.
+
+We have the following API requirements for our `Online Shopping` site from the above usecases:
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+| user creates a shopping cart  | PUT /shoppingCarts/{userId}          | create or reset shopping cart  |
+| user adds items to cart       | POST /shoppingCarts/{userId}/items | updates shopping cart  |
+| user removes an item from the cart | DEL /shoppingCarts/{userId}/items/{itemId} | removes an item from cart |
+| user submits the order | POST /users/{userId}/orders | returns order object  |
+| user's list of all orders | GET /users/{userId}/orders | returns all orders linked with user |
+
+- `Recommendation service`
+  - In the example, a [recommender application](https://github.com/strongloop/loopback4-example-shopping/tree/master/packages/recommender) is also defined alongside the `Shopping application` to mock shopping recommendation APIs. This also gives an apt usecase to demonstrate invoking external APIs.
+
+## Exposing the API endpoints
+
+We can now start creating endpoints using the api specs in the above section. Developers typically create server-side "handlers" for each route. The handlers will have to accept incoming request, parse and validate.
+
+> LoopBack develepers can configure simple javascript methods as remote endpoints. Out-of-the-box parsing and validation is done using the provided specification.
+
+Remote methods are defined in [controller classes](https://loopback.io/doc/en/lb4/Controllers.html#routing-to-controllers) and configured as API endpoints with api `decorators`.
+
+> LoopBack has inbuilt decorators to indicate api specification and the expect arguments for the remote method.
+
+For Example,
+- `login()` method in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) class is defined as `/users/login` API endpoint.
+-  argument of method `findById` in the [UserController](https://github.com/strongloop/loopback4-example-shopping/blob/master/packages/shopping/src/controllers/user.controller.ts) is decorated with `@param.path.string('userId')` which means that the userId parameter in the URL path is passed into the method at runtime.
+
+## API Security
+
+> Security implementations in Loopback can be created as separate [Authentication components](https://loopback.io/doc/en/lb4/Loopback-component-authentication.html) and the `authenticate` decorator can be used to define the authentication strategy of a particular endpoint.
```

Comment:
There is only `one`  authentication component in LoopBack 4.

But it allows users to define their own custom authentication strategies and register them as extensions to a specific extension point.

Reply:
agreed, I will change the name as authentication strategies